app/services/email_service.rb

```diff
@@ -2,29 +2,30 @@
 
 class EmailService
   def self.instance(params)
-    offender = OffenderService.get_offender(params[:nomis_offender_id])
-    pom = PrisonOffenderManagerService.get_pom(
-      params[:prison],
-      params[:primary_pom_nomis_id]
-    )
-    last_allocation = AllocationService.last_allocation(params[:nomis_offender_id])
     message = params[:message]
+    allocation = params[:allocation]
 
-    new(offender, pom, last_allocation, message)
+    new(allocation, message)
   end
 
-  def initialize(offender, pom, last_allocation, message)
-    @offender = offender
-    @pom = pom
-    @last_allocation = last_allocation
+  def initialize(allocation, message)
     @message = message
+    @allocation = allocation
+
+    @offender = OffenderService.get_offender(@allocation[:nomis_offender_id])
+    @pom = PrisonOffenderManagerService.get_pom(
+      @allocation.prison,
+      @allocation.primary_pom_nomis_id
+    )
   end
 
   def send_allocation_email
```

Comment:
Maybe `deliver_email` might be more generic than `send_allocation_email`?

Reply:
Fixed now @TobyRet